Add telemetry notice

ellismg opened the pull request:
The uses `core.notice` to print the telemetry notification as part of the setup action. We do this explicitly because we don't run `install-azd.sh` when using this action, so we don't get the telemetry notice behavior for free.

The output from the step now looks like this:

<img width="2319" alt="image" src="https://github.com/Azure/setup-azd/assets/9602953/5162fc50-357d-4c96-9615-4f95a449c949">

ellismg committed:
Add telemetry notice

hemarina reviewed (approved):
LGTM. Would you mind also add it in Readme? Thanks @ellismg for adding telemetry notice.

ellismg committed:
Respond to Marina's Feedback

- Add data collection notice to README.md (took the prose we had in
  azure/azure-dev)

- Remove `test` script, we don't presently have any JS unit level
  tests for this action, we just have a single end to end test that
  uses the action to install `azd`
